### ISSUE TITLE
CS0103 - The name 'CommandHandler' does not exist in the current context

### ISSUE THREAD

**Issue by gazsux** [PaulusParssinen/Shockky#5]
Is any body else getting this when they attempt to start Shockky.Sandbox? I've run VS with admin, installed optional .NET development tools. What am I missing?

**Comment by PaulusParssinen** [comment]
Hey, sorry for the incomplete state of the current project. This error is entirely my fault. I just recently started working on making the branches build succesfully again so I'll ping this issue once we get successful build in the wip or master branch.

**Comment by gazsux** [comment]
Hey no need to apologize, not like you're getting paid for this. I just happened upon your project and tried to build it. Been messing around with DCR files. 

By the way, do you know if there are any projects out there working on running DCR/SWF files in browser without the aid of extensions or modified browsers?

**Comment by PaulusParssinen** [comment]
Only project that has a goal of running in browsers without extensions/custom browsers exist for Adobe Flash Player (SWF) files, check out https://ruffle.rs. It's been in development for a while and still has long way to go (see https://github.com/ruffle-rs/ruffle/wiki/Roadmap). 

There is no such project (yet?) for Shockwave movies. There's two incredible projects, [ScummVM](https://github.com/scummvm/scummvm/tree/master/engines/director) and [earthquake-rs](https://github.com/csnover/earthquake-rust/tree/wip) by @csnover (wip branch) that have goal of running Shockwave movies but neither have made running files in browser an explicit goal AFAIK. If I knew Rust well enough I'd definitely try to contribute to earthquake-rs as the code is **really great** and in my opinion the only project that could see a WebAssembly-platform target in very far future. However the knowledge about the file-format and how director works is still scattered in couple projects so it's very hard for a random developer to jump in and try to contribute.

EDIT: Have you tried to input your files to https://github.com/ProjectorRays/ProjectorRays yet? It's in way better condition than this project at the moment 😅 

TLDR; SWF, some are supported. see ruffle-rs. Shockwave (DCR etc.) nope :(

**Comment by gazsux** [comment]
Ya know I actually came across Ruffle a while ago now that you bring it up. It really is fascinating what they've accomplished. 

That sucks for DCRs. If that prospect is too big for you to jump into then there's no hope for me lol. I'll just have to continue reading up and keep my eye on everyone's progress. Thanks for the insight man.
